esmf_regrid/tests/unit/experimental/unstructured_scheme/test__mesh_to_MeshInfo.py

```diff
@@ -0,0 +1,55 @@
+"""Unit tests for :func:`esmf_regrid.experimental.unstructured_scheme._mesh_to_MeshInfo`."""
+
+from iris.coords import AuxCoord
+from iris.experimental.ugrid import Connectivity, Mesh
+import numpy as np
+from numpy import ma
+import scipy.sparse
+
+from esmf_regrid.esmf_regridder import Regridder
+from esmf_regrid.experimental.unstructured_scheme import _mesh_to_MeshInfo
+
+
+def _example_mesh():
+    """Generate a global mesh with a square pyramid topology."""
+    fnc_array = [
+        [0, 1, 2, 3],
+        [0, 1, 4, -1],
+        [1, 2, 4, -1],
+        [2, 3, 4, -1],
+        [3, 0, 4, -1],
+    ]
+    fnc_mask = [
+        [0, 0, 0, 0],
+        [0, 0, 0, 1],
+        [0, 0, 0, 1],
+        [0, 0, 0, 1],
+        [0, 0, 0, 1],
+    ]
+    fnc_ma = ma.array(fnc_array, mask=fnc_mask)
+    fnc = Connectivity(
+        fnc_ma,
+        cf_role="face_node_connectivity",
+        start_index=0,
+    )
+    lon_values = [120, 120, -120, -120, 0]
+    lat_values = [-60, 60, 60, -60, 0]
+    lons = AuxCoord(lon_values, standard_name="longitude")
+    lats = AuxCoord(lat_values, standard_name="latitude")
+    mesh = Mesh(2, ((lons, "x"), (lats, "y")), fnc)
+    return mesh
```

Comment:
Polygons are described in such a way that nodes are traversed clockwise. There is still some question about what orientations UGRID allows and what orientations ESMF can handle. If UGRID can allow more than ESMF can handle, we may have to catch and handle such cases ourselves.

Reply:
Having looked at the UGRID conventions again, the only case in which both clockwise and anticlockwise orientations are allowed is for [Fully 3D unstructured meshes](https://ugrid-conventions.github.io/ugrid-conventions/#fully-3d-unstructured-ie-non-layered-mesh-topology). In all other cases, anticlockwise orientation is specified.